packages/gatsby-theme-docs/src/overrides/topbar-login-info.js

```diff
@@ -1,10 +1,16 @@
 import { useContext } from 'react';
 import ConfigContext from '../components/config-context';
 import UserProfile from '../modules/sso/components/avatar';
-
+import { LearningContextState } from '../modules/self-learning/components/learning-context';
 const LoginInfo = () => {
   const { hideLogin } = useContext(ConfigContext);
-  return hideLogin ? null : <UserProfile />;
+  const {
+    user: { profile },
+  } = useContext(LearningContextState);
+  if (hideLogin && profile) return <UserProfile />;
+  if (hideLogin && !profile) return null;
+
+  return <UserProfile />;
```

Comment:
can't help but thinking this could be done as a simpler coditional.  showing the user profile is the default and the only exception is to not show it if not logged and and hiding it. 
```suggestion
  return (hideLogin && !profile) ? null: <UserProfile />;
```